### PR TITLE
Resolve scopes from authenticated client sessions when selecting attributes

### LINKED ISSUE #35192: Resolve scopes from authenticated client sessions when selecting attributes
### Description

Support for enabling and marking attributes a required based on client scopes when using the account console.

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by pedroigor**
Closes #35192

### CODE REVIEW

File: services/src/main/java/org/keycloak/services/DefaultKeycloakContext.java [hunk 1]
```diff
@@ -113,6 +116,11 @@ public void setRealm(RealmModel realm) {
 
     @Override
     public ClientModel getClient() {
+        if (client == null) {
```
Comment (pedroigor) [inline services/src/main/java/org/keycloak/services/DefaultKeycloakContext.java:119]: If it makes sense, I can add a check to the client set by `setClient` is always the same from the authentication session, if the latter is set.

File: server-spi/src/main/java/org/keycloak/models/KeycloakContext.java [hunk 2]
```diff
@@ -104,4 +104,8 @@ default Locale resolveLocale(UserModel user, Theme.Type themeType) {
     void setHttpRequest(HttpRequest httpRequest);
 
     void setHttpResponse(HttpResponse httpResponse);
+
+    UserSessionModel getUserSession();
```
Comment (pedroigor) [inline server-spi/src/main/java/org/keycloak/models/KeycloakContext.java:108]: We already have the `realm`, `client`, and `authenticationSession` available from the context. It should be useful to also expose the `UserSessionModel` associated with the current session/request.

In this particular PR, we use the user session to resolve client scopes from authenticated client sessions.

**Review by sguilhen — APPROVED** [review]
Changes make sense to me, thanks @pedroigor !

**Review by ahus1 — APPROVED** [review]
Approving based on @sguilhen's review

**Pull request merged**